### PR TITLE
Add current_score to ModelCheckpoint.on_save_checkpoint

### LINKED ISSUE #4712: Add current_model_score to ModelCheckpoint.on_save_checkpoint
## 🚀 Feature

Add `current_model_score` to `ModelCheckpoint.on_save_checkpoint`

https://github.com/PyTorchLightning/pytorch-lightning/blob/c208ac68c88a23977ae3d156dabb1440a9f1c9f3/pytorch_lightning/callbacks/model_checkpoint.py#L186-L191

### Motivation

Say you have 3 checkpoints saved (1, 2, 3). You got your best score on (1). It is a minimization task.

This would be each checkpoint saved state:
```
[
    ('epoch=1.ckpt', {
        'monitor': 'foo',
        'best_model_score': tensor(0.1),
        'best_model_path': 'epoch=1.ckpt'
    }),
    ('epoch=2.ckpt', {
        'monitor': 'foo',
        'best_model_score': tensor(0.1),
        'best_model_path': 'epoch=1.ckpt'
    }),
    ('epoch=3.ckpt', {
        'monitor': 'foo',
        'best_model_score': tensor(0.1),
        'best_model_path': 'epoch=1.ckpt'
    })
]
```
Notice how there is no info about the actual score for that particular checkpoint.

### Pitch

Add the actual score for each checkpoint state

```
[
    ('epoch=1.ckpt', {
        'monitor': 'foo',
        'best_model_score': tensor(0.1),
        'best_model_path': 'epoch=1.ckpt'
        'current_model_score': tensor(0.1),
    }),
    ('epoch=2.ckpt', {
        'monitor': 'foo',
        'best_model_score': tensor(0.1),
        'best_model_path': 'epoch=1.ckpt'
        'current_model_score': tensor(0.2),
    }),
    ('epoch=3.ckpt', {
        'monitor': 'foo',
        'best_model_score': tensor(0.1),
        'best_model_path': 'epoch=1.ckpt'
        'current_model_score': tensor(0.3),
    })
]
```

I can work on the PR

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #4712

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [n/a] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -188,6 +189,7 @@ def on_save_checkpoint(self, trainer, pl_module) -> Dict[str, Any]:
             "monitor": self.monitor,
             "best_model_score": self.best_model_score,
             "best_model_path": self.best_model_path,
+            "current_score": self.current_score,
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/model_checkpoint.py:192]:
```suggestion
            "checkpoint_score": self.current_score,
```
maybe `checkpoint_score`?

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 2]
```diff
@@ -155,6 +155,7 @@ def __init__(
         self.period = period
         self.last_global_step_saved = -1
         self.prefix = prefix
+        self.current_score = 0
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/model_checkpoint.py:158]:
```suggestion
        self.current_score = None
```
default to `None`?

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/callbacks/model_checkpoint.py:192]: IMO thats redundant. Its in a class with checkpoint in its name.

In any case, we could do `current_model_score` to be consistent with `best_model_score`. But I find `current_score` simpler.

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/callbacks/model_checkpoint.py:158]: `best_model_score` defaults to 0. Chose it here for consistency.

Do you want me to change both to `None`?

Comment (Borda) [reply pytorch_lightning/callbacks/model_checkpoint.py]: yes, that was my thoughts that for min 0 is already the min

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 3]
```diff
@@ -155,6 +155,7 @@ def __init__(
         self.period = period
         self.last_global_step_saved = -1
         self.prefix = prefix
+        self.current_score = None
         self.best_k_models = {}
         self.kth_best_model_path = ""
         self.best_model_score = 0
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:161]:
```suggestion
        self.best_model_score = None
```

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/callbacks/model_checkpoint.py]: see: https://github.com/PyTorchLightning/pytorch-lightning/pull/4721/files#r525591032

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 4]
```diff
@@ -563,11 +565,14 @@ def _update_best_and_save(
             self.best_k_models.pop(self.kth_best_model_path)
             del_list.append(delpath)
 
-        # do not save non, for replace then by +/- inf
+        # do not save nan, replace with +/- inf
         if torch.isnan(current):
-            current = {"min": torch.tensor(float('inf')), "max": torch.tensor(-float('inf'))}[self.mode]
+            current = {"min": torch.tensor(float('inf')), "max": torch.tensor(float('-inf'))}[self.mode]
```
Comment (s-rog) [inline pytorch_lightning/callbacks/model_checkpoint.py:570]: mind including a test for this as well?

Comment (carmocca) [reply pytorch_lightning/callbacks/model_checkpoint.py:570]: Done

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/callbacks/model_checkpoint.py:192]: cool

**Review by s-rog — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Good idea !
